homeassistant/components/sensor/sht31.py

```
@@ -0,0 +1,152 @@
+Support for Sensirion SHT31 temperature and humidity sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.sht31/
+"""
+
+from datetime import timedelta
+import logging
+import math
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    TEMP_CELSIUS, CONF_NAME, CONF_MONITORED_CONDITIONS)
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.temperature import display_temp
+from homeassistant.const import PRECISION_TENTHS
+from homeassistant.util import Throttle
+
+
+REQUIREMENTS = ['Adafruit-GPIO==1.0.3',
+                'Adafruit-SHT31==1.0.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_I2C_ADDRESS = 'i2c_address'
+
+DEFAULT_NAME = 'SHT31'
+DEFAULT_I2C_ADDRESS = 0x44
+
+SENSOR_TEMPERATURE = 'temperature'
+SENSOR_HUMIDITY = 'humidity'
+SENSOR_TYPES = (SENSOR_TEMPERATURE, SENSOR_HUMIDITY)
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS):
+        vol.All(vol.Coerce(int), vol.Range(min=0x44, max=0x45)),
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSOR_TYPES)):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    from Adafruit_SHT31 import SHT31
+
+    i2c_address = config.get(CONF_I2C_ADDRESS)
+    sensor = SHT31(address=i2c_address)
+
+    try:
+        if sensor.read_status() is None:
+            raise ValueError("CRC error while reading SHT31 status")
+    except (OSError, ValueError):
+        raise HomeAssistantError("SHT31 sensor not detected at address %s " %
```

I don't think we should raise this error here. Log an error and return.